### ISSUE TITLE
Excessive memory usage and eventual crash 

### ISSUE THREAD

**Issue by Tripplesixty** [rakshasa/rtorrent#272]
I've been experiencing excessive memory usage with rtorrent/libtorrent lately.  Over the course of a day rtorrent will consume all my memory and swap ~24GB in total. I'm running version 0.9.4/0.13.4 and have been for the last year without problems until now.  The last time I caught it at peak memory consumption I noticed rtorrent being filled with the following message, repeated many times:

Scheduled command failed: erasedata: ExecFile::execute(...) Fork failed.

Is this something obvious or do I need to core rtorrent for further inspection?  Assuming I do that would I need a debug build? Any suggestions would be helpful! Just let me know what logs/files I can provide.

**Comment by Tripplesixty** [comment]
So I am running ubuntu 14.04 `Linux server 3.13.0-24-generic #47-Ubuntu SMP Fri May 2 23:30:00 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux`.  I compiled from source with support for rutorrent (xmlrpc) given that I compiled nearly a year ago I don't remember the exact parameters.  I am now experiencing a crash almost every night.  When I look for the process in the morning rtorrent is gone.  This time around I started within GDB to try and catch the crash, however I'm not running any kind of debug build.  I've attached my rc file below. I'll report back when it crashes again, likely within the next 24 hours.  I can attach more logs this time around.

rtorrent.rc http://pastebin.com/7MiZJ5dQ

**Comment by Tripplesixty** [comment]
The rtorrent process died last night with following output in the screen terminal in gdb.  Logs will be attached.
```
( 0:01:43) Scheduled command failed: autowatch<username>: ExecFile::execute(...) Fork failed.
( 0:01:48) Scheduled command failed: autowatch<username>: ExecFile::execute(...) Fork failed.
( 0:01:53) Scheduled command failed: autowatch<username>: ExecFile::execute(...) Fork failed.
( 0:02:48) Scheduled command failed: autowatch<username>: ExecFile::execute(...) Fork failed.
( 0:02:53) Scheduled command failed: autowatch<username>: ExecFile::execute(...) Fork failed.
( 0:02:58) Scheduled command failed: autowatch<username>: ExecFile::execute(...) Fork failed.
( 0:03:43) Scheduled command failed: autowatch<username>: ExecFile::execute(...) Fork failed.
( 0:03:48) Scheduled command failed: autowatch<username>: ExecFile::execute(...) Fork failed.
( 0:03:53) Scheduled command failed: autowatch<username>: ExecFile::execute(...) Fork failed.
( 0:03:58) Scheduled command failed: autowatch<username>: ExecFile::execute(...) Fork failed.
( 0:03:58) Scheduled command failed: erasedata<username>: ExecFile::execute(...) Fork failed.
( 0:04:43) Scheduled command failed: autowatch<username>: ExecFile::execute(...) Fork failed.
( 0:04:48) Scheduled command failed: autowatch<username>: ExecFile::execute(...) Fork failed.
( 0:04:53) Scheduled command failed: autowatch<username>: ExecFile::execute(...) Fork failed.
( 0:05:58) Scheduled command failed: autowatch<username>: ExecFile::execute(...) Fork failed.
( 0:05:58) Scheduled command failed: erasedata<username>: ExecFile::execute(...) Fork failed.
( 0:06:03) Scheduled command failed: autowatch<username>: ExecFile::execute(...) Fork failed.
( 0:06:08) Scheduled command failed: autowatch<username>: ExecFile::execute(...) Fork failed.
( 0:06:13) Scheduled command failed: erasedata<username>: ExecFile::execute(...) Fork failed.
( 0:06:13) Scheduled command failed: autowatch<username>: ExecFile::execute(...) Fork failed.
( 0:06:18) Scheduled command failed: autowatch<username>: ExecFile::execute(...) Fork failed.
( 0:06:23) Scheduled command failed: autowatch<username>: ExecFile::execute(...) Fork failed.
( 0:06:25) Scheduled command failed: autowatch<username>: ExecFile::execute(...) Fork failed.
( 0:06:25) Scheduled command failed: erasedata<username>: ExecFile::execute(...) Fork failed.
      
Program terminated with signal SIGKILL, Killed.
The program no longer exists.
```
Sanitized Logs are attached:
Storage Logs: https://drive.google.com/file/d/0BwiIZfw4AJ8MSzFlZ0x2anYtY28/view?usp=sharing
Debug Logs: https://drive.google.com/file/d/0BwiIZfw4AJ8MMWQ0T2lxRVJLOG8/view?usp=sharing

**Comment by Tripplesixty** [comment]
Here is an updated view of the process memory... this is sampled at every second over the last 24 hours.  As you can see sometime late in the morning the memory starts to rise exponentially.  Again this crash is happening every night.
http://imgbox.com/Li2Y7zYO

**Comment by Tripplesixty** [comment]
I believe this bug is related to a faulty client.  See this ticket https://github.com/SiCKRAGETV/SickRage/pull/585.  It has since been fixed and I have upgraded this piece of software and haven't experienced this bug since.

**Comment by Papanikles** [comment]
I have same problem. How apply the fix?

The suggested fix is to rtorrent in python. My rtorrent/libttorrent is from official site and i think is in C/C++.

**Comment by Tripplesixty** [comment]
@Papanikles my problem was caused by 3rd party software. The patch was applied to SickrageTV software.

**Comment by Papanikles** [comment]
@Tripplesixty thank you for your reply.

And don't have any patch for original source?

**Comment by Tripplesixty** [comment]
@Papanikles - Please see the aforementioned link to the ticket in the SickRage repository: https://github.com/SiCKRAGETV/SickRage/pull/585

**Comment by Papanikles** [comment]
I had seen. The problem is that the patch is in Python (.py) and the original source is in C / C ++. So this patch doesn't work on: http://libtorrent.rakshasa.no/

**Comment by Tripplesixty** [comment]
Like I said before.  The patch that I applied was to the 3rd party software that acted as a client to rtorrent. rtorrent itself was never patched when I fixed this.  I would suggest making sure you understand what your problem is and if you are running any similar third party software.  There is no C/C++ fix for this a this time.

**Comment by PanDoreS** [comment]
Hello, I'm running CentOS Linux release 7.3.1611 (Core) and it's been a few months since I have exactly the same issue. After 2/3 weeks of usage, rTorrent shows me those messages. It result with a high RAM usage (memory leak ?) that keeps growing, and after some time, the server become laggy and crash.
I don't have any 3rd party software client on rTorrent, except ruTorrent (v3.8).
A friend who have a different server but the same CentOS version on it have the same issue as me.
I've also compiled the feature-bind branch, same result.
I'm short of ideas ... 
Here's my .rtorrent config : [https://pastebin.com/c2rEWVGL](url)
I've just added the log output on "info" level.
I'll keep this post updated with what I can find.

Thanks
